Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\GeneratePerimeter\Strategy;
+
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
+use OpenOrchestra\BaseBundle\Context\CurrentSiteIdInterface;
+
+/**
+ * Class NodeGeneratePerimeterStrategy
+ */
+class NodeGeneratePerimeterStrategy implements GeneratePerimeterStrategyInterface
+{
+    protected $nodeRepository;
+    protected $contextManager;
+
+    /**
+     * @param NodeRepositoryInterface $nodeRepository
+     */
+    public function __construct(
+        NodeRepositoryInterface $nodeRepository,
+        CurrentSiteIdInterface $contextManager
+
+    ) {
+        $this->nodeRepository = $nodeRepository;
+        $this->contextManager = $contextManager;
+    }
+
+    /**
+     * Return the supported perimeter type
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return NodeInterface::ENTITY_TYPE;
+    }
+
+    /**
+     * Generate perimeter
+     *
+     * @return array
+     */
+    public function generatePerimeter()
+    {
+        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_NODE_ID);
+        $treeNodes = $this->formatPerimeter($treeNodes[0]);
+
+        return $treeNodes;
+    }
+
+    /**
+     * get perimeter configuration
+     *
+     * @return array
+     */
+    public function getPerimeterConfiguration()
+    {
+        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_NODE_ID);
+        $treeNodes = $this->formatConfiguration($treeNodes[0]);
+
+        return $treeNodes;
+    }
+
+    /**
+     * format perimeter
+     *
+     * @param array $treeNodes
+     *
+     * @return array
+     */
+    protected function formatPerimeter(&$treeNodes)
+    {
+        $path = $treeNodes['node']['path'];
```

Comment:
The result come from repository 'findTreeNode'. So, if there is a 0 index, there is a 'node' index. On my point of view, i should just test 'path'?